### ISSUE TITLE
Build failure: amdvlk

### ISSUE THREAD

**Issue by pagedMov** [NixOS/nixpkgs#348903]
### Steps To Reproduce

1. Attempt to build the steam package
2. Wait 10 minutes for amdvlk to compile

### Build log

```
building the system configuration...
trace: evaluation warning: pagedmov profile: Setting programs.eza.icons to a Boolean is deprecated.
Please update your configuration so that

  programs.eza.icons = "auto"
[2/17/72 built, 9/34/58 copied (575.4/1708.8 MiB), 122.8/325.3 MiB DL] building amdvlk-2024.Q3.2 (buildPhase): [1447/2841] Building CXX object compiler/llpc/llvm/lib/CodeGeerror: builder for '/nix/store/jy4xs329wkwznisac8p52sd7n73fxqb9-amdvlk-2024.Q3.2.drv' failed with exit code 1;
       last 10 log lines:
       > Compiling (dxc spirv=True) InitExecuteIndirect.hlsl:InitExecuteIndirect -> InitExecuteIndirect_spv.h...Success
       > Compiling (dxc spirv=True) PairCompression.hlsl:PairCompression -> PairCompression_spv.h...Success
       > Compiling (dxc spirv=True) MergeSort.hlsl:MergeSort -> MergeSort_spv.h...Success
       > Compiling (dxc spirv=True) BuildFastAgglomerativeLbvh.hlsl:BuildFastAgglomerativeLbvh -> BuildFastAgglomerativeLbvh_spv.h...Success
       > Compiling (dxc spirv=True) BuildBVHTDTR.hlsl:BuildBVHTD -> BuildBVHTDTR_spv.h...Success
       > Compiling (dxc spirv=True) BuildQBVH.hlsl:BuildQBVH -> BuildQBVH_spv.h...Success
       > Compiling (dxc spirv=True) BuildParallel.hlsl:BuildBvh -> BuildParallel_spv.h...Success
       > [1821/2841] Building AMDGPUGenAsmWriter.inc...
       > [1822/2841] Building AMDGPUGenAsmMatcher.inc...
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/jy4xs329wkwznisac8p52sd7n73fxqb9-amdvlk-2024.Q3.2.drv'.
error: 1 dependencies of derivation '/nix/store/l1bip6xf2mrjdjchpp48nwdjf02h86qv-graphics-drivers-32bit.drv' failed to build
error: 1 dependencies of derivation '/nix/store/bscsr3pmf8dn3vlsw50zim95b1n6n5j2-steam-run-usr-multi.drv' failed to build
error: 1 dependencies of derivation '/nix/store/dj5mbykaqq6yyxrivnma9p7anscxw4cg-steam-usr-multi.drv' failed to build
error: 1 dependencies of derivation '/nix/store/gqwvvg5zvjqxqji8px0l7fhcj3671dgg-graphics-driver.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/iv62pkx4480fib4nd9qv70hvpir9rm5h-steam-fhs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rqjhi64pcp6yqq5dv157z9ypszg538s1-steam-run-fhs.drv' failed to build
error: 1 dependencies of derivation '/nix/store/s5ihzjmvl0xl1clcgfvpdaz67bb1mbyp-steam-bwrap.drv' failed to build
error: 1 dependencies of derivation '/nix/store/l4gi74lnk8y11yj3s2nnyj9qb4gx933w-steam-run-bwrap.drv' failed to build
error: 1 dependencies of derivation '/nix/store/48pggdqw9vc60pvd05jzgpjv3jwkxm72-tmpfiles.d.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hrcafmxkyvbr3xrm4frrn3683v87k03i-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/d94dryr5wrmpfmjy76k9f6szv2njawg7-steam-run.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xhcfbyzda7hc3i5z6awin90j8p4znhl9-steam.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rwrh1ndpcvlqh0sn50ahpp7nyjmmbxdr-nixos-system-argon-24.11.20241014.a3c0b3b.drv' failed to build

Compiler build log:
https://gist.github.com/pagedMov/2fb0b5eb84588461ef7222e0e7c4e9ef
```

### Additional context

This was working up until today, and I'm not sure what exactly happened. I was able to build my system config with steam last night, woke up this morning and amdvlk refuses to build now. I'm enabling steam via the programs.steam module, tried it in both my system and home-manager configurations with no change in the behavior. Tried git bisect, and commits that were previously able to build the package no longer can.

### Notify maintainers

@Flakebi

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.2, NixOS, 24.11 (Vicuna), 24.11.20241009.5633bcf`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"home-manager, nixos-24.05"`
 - nixpkgs: `/nix/store/60sn02zhawl3kwn0r515zff3h6hg6ydz-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by pagedMov** [comment]
Appears to be related to the hardware.amdgpu.amdvlk attribute set, disabling it allowed my config to rebuild. Currently testing it
with amdgpu.amdvlk.support32Bit enabled, will report results

**Comment by pagedMov** [comment]
Confirmed fails to build even with the support32Bit option enabled

**Comment by oluceps** [comment]
Same

**Comment by oluceps** [comment]
https://github.com/NixOS/nixpkgs/pull/338410

**Comment by unhitched9271** [comment]
Im experiencing this too (identical error), however I'm experiencing it when building various apps like bitwarden-desktop for example - can provide logs if required

**Comment by RedJ1ve** [comment]
I'm also having the exact same problem

**Comment by Ate329** [comment]
I'm getting exactly the same error as well

**Comment by tijso** [comment]
Same issue for me just started after updating. Any fix or workaround or just a waiting game?

**Comment by SeekerOfAsh** [comment]
I am experiencing the same issue when trying to build my system. AMDVLK fails to build.
```
error: builder for '/nix/store/h2adl88hvwrndpq24kw8pav8hwzia1yb-amdvlk-2024.Q3.2.drv' failed with exit code 1;
       last 10 log lines:
       > [1826/2845] Building AMDGPUGenAsmMatcher.inc...
       > [1827/2845] Building CXX object compiler/llpc/llvm/lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/DAGCombiner.cpp.o
       > /build/amdvlk-src/drivers/llvm-project/llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp: In function 'llvm::SDValue FoldIntToFPToInt(llvm::SDNode*, llvm::SelectionDAG&)':
       > /build/amdvlk-src/drivers/llvm-project/llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp:17939:23: warning: possibly dangling reference to a temporary [-Wdangling-reference]
       > 17939 |   const fltSemantics &Sem = N0.getValueType().getFltSemantics();
       >       |                       ^~~
       > /build/amdvlk-src/drivers/llvm-project/llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp:17939:62: note: the temporary was destroyed at the end of the full expression 'N0.llvm::SDValue::getValueType().llvm::EVT::getFltSemantics()'
       > 17939 |   const fltSemantics &Sem = N0.getValueType().getFltSemantics();
       >       |                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/h2adl88hvwrndpq24kw8pav8hwzia1yb-amdvlk-2024.Q3.2.drv'.
```

**Comment by Svenum** [comment]
Same here:
```bash
building the system configuration...
error: builder for '/nix/store/jy4xs329wkwznisac8p52sd7n73fxqb9-amdvlk-2024.Q3.2.drv' failed with exit code 1;
       last 25 log lines:
       > Compiling (dxc spirv=True) InitExecuteIndirect.hlsl:InitExecuteIndirect -> InitExecuteIndirect_spv.h...Success
       > Compiling (dxc spirv=True) DeserializeAS.hlsl:DeserializeAS -> DeserializeAS_spv.h...Success
       > Compiling (dxc spirv=True) PairCompression.hlsl:PairCompression -> PairCompression_spv.h...Success
       > Compiling (dxc spirv=True) InitAccelerationStructure.hlsl:InitAccelerationStructure -> InitUpdateAccelerationStructure_spv.h...Success
       > Compiling (dxc spirv=True) InitAccelerationStructure.hlsl:InitAccelerationStructure -> InitAccelerationStructure_spv.h...Success
       > Compiling (dxc spirv=True) MergeSort.hlsl:MergeSort -> MergeSort_spv.h...Success
       > Compiling (dxc spirv=True) BuildBVHTDTR.hlsl:BuildBVHTD -> BuildBVHTDTR_spv.h...Success
       > Compiling (dxc spirv=True) CompactAS.hlsl:CompactAS -> CompactAS_spv.h...Success
       > Compiling (dxc spirv=True) BuildFastAgglomerativeLbvh.hlsl:BuildFastAgglomerativeLbvh -> BuildFastAgglomerativeLbvh_spv.h...Success
       > Compiling (dxc spirv=True) BuildQBVH.hlsl:BuildQBVH -> BuildQBVH_spv.h...Success
       > Compiling (dxc spirv=True) Rebraid.hlsl:Rebraid -> Rebraid_spv.h...Success
       > Compiling (dxc spirv=True) BuildParallel.hlsl:BuildBvh -> BuildParallel_spv.h...Success
       > [1822/2841] Building CXX object compiler/llpc/llvm/lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/FastISel.cpp.o
       > [1823/2841] Building CXX object compiler/llpc/llvm/lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/LegalizeDAG.cpp.o
       > [1824/2841] Building AMDGPUGenAsmWriter.inc...
       > [1825/2841] Building AMDGPUGenAsmMatcher.inc...
       > [1826/2841] Building CXX object compiler/llpc/llvm/lib/CodeGen/SelectionDAG/CMakeFiles/LLVMSelectionDAG.dir/DAGCombiner.cpp.o
       > /build/amdvlk-src/drivers/llvm-project/llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp: In function 'llvm::SDValue FoldIntToFPToInt(llvm::SDNode*, llvm::SelectionDAG&)':
       > /build/amdvlk-src/drivers/llvm-project/llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp:17939:23: warning: possibly dangling reference to a temporary [-Wdangling-reference]
       > 17939 |   const fltSemantics &Sem = N0.getValueType().getFltSemantics();
       >       |                       ^~~
       > /build/amdvlk-src/drivers/llvm-project/llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp:17939:62: note: the temporary was destroyed at the end of the full expression 'N0.llvm::SDValue::getValueType().llvm::EVT::getFltSemantics()'
       > 17939 |   const fltSemantics &Sem = N0.getValueType().getFltSemantics();
       >       |                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/jy4xs329wkwznisac8p52sd7n73fxqb9-amdvlk-2024.Q3.2.drv'.
error: 1 dependencies of derivation '/nix/store/l1bip6xf2mrjdjchpp48nwdjf02h86qv-graphics-drivers-32bit.drv' failed to build
error: 1 dependencies of derivation '/nix/store/yn18fpg3k46nikbz6vzn85kg1wvwfa3h-graphics-driver.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ln880b7vk1kg0w6ljkwbfyzr7mrnkdak-tmpfiles.d.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rfh8k1wapwx9f1wik9p3hixwygs04dfm-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vsn0sqj99hg8xqqvan1b6cr5pmhinvby-nixos-system-Yon-24.11.20241014.a3c0b3b.drv' failed to build
```

**Comment by Kim-Ronin** [comment]
same here.... not just with amdvlk, but with multiple packages. Commenting them in config will lead to amdvlk



error: builder for '/nix/store/jy4xs329wkwznisac8p52sd7n73fxqb9-amdvlk-2024.Q3.2.drv' failed with exit code 1;
       last 10 log lines:
       > Compiling (dxc spirv=True) PairCompression.hlsl:PairCompression -> PairCompression_spv.h...Success
       > Compiling (dxc spirv=True) MergeSort.hlsl:MergeSort -> MergeSort_spv.h...Success
       > Compiling (dxc spirv=True) BuildFastAgglomerativeLbvh.hlsl:BuildFastAgglomerativeLbvh -> BuildFastAgglomerativeLbvh_spv.h...Success
       > Compiling (dxc spirv=True) Update.hlsl:UpdateTriangles -> UpdateTriangles_spv.h...Success
       > Compiling (dxc spirv=True) BuildQBVH.hlsl:BuildQBVH -> BuildQBVH_spv.h...Success
       > Compiling (dxc spirv=True) BuildBVHTDTR.hlsl:BuildBVHTD -> BuildBVHTDTR_spv.h...Success
       > Compiling (dxc spirv=True) BuildParallel.hlsl:BuildBvh -> BuildParallel_spv.h...Success
       > [1820/2841] Building AMDGPUGenAsmWriter.inc...
       > [1821/2841] Building AMDGPUGenAsmMatcher.inc...
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/jy4xs329wkwznisac8p52sd7n73fxqb9-amdvlk-2024.Q3.2.drv'.
error: 1 dependencies of derivation '/nix/store/2wia0900dnqv62r5h0ys06417f988g4f-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4s26avigpm7bdxz5fh2si5qmi28bwwij-nixos-system-Terminal-416-24.11.20241014.a3c0b3b.drv' failed to build

**Comment by unkn** [comment]
Seems to be caused by https://github.com/NixOS/nixpkgs/commit/c3948c21ef00fbf8ef02f3c2922e84ba72e8a62b
It also compiles with glslang-15.0.0 (with shared libs), but haven't tested it yet.

**Comment by bendlas** [comment]
see also: https://github.com/NixOS/nixpkgs/pull/344568#issuecomment-2409029219

**Comment by LoganBarnett** [comment]
In addition to @unkn's overlay example above, I added this in the new attrset for `my_gslang`:

``` nix
    cmakeFlags = [];
```

This effectively reverses the mentioned commit https://github.com/NixOS/nixpkgs/commit/c3948c21ef00fbf8ef02f3c2922e84ba72e8a62b.  I had to do this to get it to work again.  I didn't test if the rest of the overlay was necessary for me after adding this line.

**Comment by Flakebi** [comment]
I opened a PR to fix glslang until staging-next is merged:  #349246
